zerver/lib/user_groups.py

```diff
@@ -540,3 +541,23 @@ def get_system_user_group_for_user(user_profile: UserProfile) -> UserGroup:
         name=system_user_group_name, realm=user_profile.realm, is_system_group=True
     )
     return system_user_group
+
+
+def get_server_supported_permission_settings() -> ServerSupportedPermissionSettings:
+    realm_permisison_group_settings: Dict[str, GroupPermissionSetting] = {}
+    for permission_name, permission_config in Realm.REALM_PERMISSION_GROUP_SETTINGS.items():
+        realm_permisison_group_settings[permission_name] = permission_config
+
+    stream_permisison_group_settings: Dict[str, GroupPermissionSetting] = {}
+    for permission_name, permission_config in Stream.stream_permission_group_settings.items():
+        stream_permisison_group_settings[permission_name] = permission_config
+
+    group_permisison_settings: Dict[str, GroupPermissionSetting] = {}
+    for permission_name, permission_config in UserGroup.GROUP_PERMISSION_SETTINGS.items():
+        group_permisison_settings[permission_name] = permission_config
+
+    return ServerSupportedPermissionSettings(
+        realm=realm_permisison_group_settings,
+        stream=stream_permisison_group_settings,
+        group=group_permisison_settings,
+    )
```

Comment:
permission is misspelled for several variables here.

Reply:
Fixed those.